unresolved external symbol error

vcpkg package management program version 2023-06-15-8c254a5fb6c503615834fc66bd0717664a339a2e
Microsoft Visual Studio Community 2022 (64-bit) - CurrentVersion 17.6.4
msvc console project
sqlitecpp 3.3.0


When I install the library using vcpkg and run the test code in VisualStudio, I get the following unresolved external symbol error.

```
********.obj error LNK2001: The external symbol "int const SQLite::OPEN_READONLY" (?OPEN_READONLY@SQLite@@@3HB) is unresolved
********.exe : fatal error LNK1120: 1 unresolved external reference
```

How can I resolve this?

If only the sqlite3 library is used from vcpkg, there is no error.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -215,7 +215,7 @@ endif (SQLITE_USE_LEGACY_STRUCT)
 
 if(BUILD_SHARED_LIBS)
     if(WIN32)
-        add_definitions("-DSQLITECPP_COMPILE_DLL")
+        target_compile_definitions(SQLiteCpp PUBLIC "SQLITECPP_COMPILE_DLL")
         target_compile_definitions(SQLiteCpp PRIVATE "SQLITECPP_DLL_EXPORT")
     endif()
 endif()

